OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -364,15 +389,56 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			}
 
 			// Too spammy to keep enabled all the time, but very useful when debugging specific issues.
-			// AIUtils.BotDebug("{0} couldn't decide what to build for queue {1}.", queue.Actor.Owner, queue.Info.Group);
+			// AIUtils.BotDebug($"{queue.Actor.Owner} couldn't decide what to build for queue {queue.Info.Group}.");
 			return null;
 		}
 
+		CPos? ChooseBarrierLocation(ActorInfo wallInfo)
+		{
+			CPos? FindUnwalledEdgeAround(Actor actor)
+			{
+				if (actor == null || actor.Info == null)
+					return null;
+
+				var buildingInfo = actor.Info.TraitInfoOrDefault<BuildingInfo>();
+				if (buildingInfo == null)
+					return null;
```

Comment:
I'm not sure this is necessary, considering you only pass an actor you got via `ActorsHavingTrait<Building>()`.

Reply:
I pass actors from a `string` array. See also:

https://github.com/OpenRA/OpenRA/blob/a7c040ff17f3fef6b8884073f901ae8535fa9081/OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs#L440-L442